### ISSUE TITLE
Efficient Kronecker Gradients in Zygote

### ISSUE THREAD

**Issue by jessebett** [MichielStock/Kronecker.jl#11]
Can you review the implementation of the gradients for `kron` in [Zygote](https://github.com/FluxML/Zygote.jl/blob/a23e76b40c4dc34e2f674c0b1222f7062bb65445/src/lib/array.jl#L231-L241) and [Tracker](https://github.com/FluxML/Tracker.jl/blob/97420eace7288fc82847ff852693bb96e3d2f84a/src/lib/array.jl#L261-L273)? I directly ported these from [Tensorflow](https://github.com/tensorflow/tensorflow/blob/23c218785eac5bfe737eec4f8081fd0ef8e0684d/tensorflow/contrib/kfac/python/ops/utils.py#L121-L127).

Specifically could you comment on the implementation, and whether it could benefit from your package?

**Comment by MichielStock** [comment]
I will look at it Friday during the hackathon!

**Comment by MichielStock** [comment]
Can we discuss irl somewhere this week? Have some questions.

Btw, `kron` from Zygote seems to work a bit faster than base.

**Comment by jessebett** [comment]
Yes. I'm around tomorrow and will also be at the hackathon on Friday working on Zygote stuff. The `kron` in `Zygote` is just some `reshape` rules. I did not benchmark against base, but that's pretty surprising, I wonder why.

**Comment by bogiebro** [comment]
The reshape/broadcasting implementation doesn't use any of the nice structure from the module. What about the following:
```julia
vec(A) = reshape(A, :)

Zygote.@adjoint kronecker(A, B) = kronecker(A,B), dy-> (
  dy.A .* (vec(dy.B)' * vec(B)), dy.B .* (vec(A)' * vec(dy.A)))
```
